IFC_Adapter/IfcAdapter.cs

```diff
@@ -0,0 +1,132 @@
+ */
+
+using BH.oM.Adapters.IFC;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.ComponentModel;
+using Xbim.Ifc;
+using Xbim.ModelGeometry.Scene;
+
+namespace BH.Adapter.IFC
+{
+    [Description("Adapter allowing to interact with the locally saved IFC files.")]
+    public partial class IfcAdapter : BHoMAdapter, IDisposable
+    {
+        /***************************************************/
+        /**** Public properties                         ****/
+        /***************************************************/
+
+        [Description("Settings of the IFC adapter.")]
+        public IfcSettings IFCSettings { get; set; } = null;
+
+
+        /***************************************************/
+        /**** Constructors                              ****/
+        /***************************************************/
+
+        [Description("Initialises the IFC Adapter with a target location.")]
+        [Input("targetLocation", "Path to the IFC file to be interacted with, including file extension.")]
+        [Input("settings", "Settings of the IFC adapter.")]
+        [Input("active", "If true, the adapter consumes the file at target location and prepares for interacting with it.")]
+        public IfcAdapter(string targetLocation, IfcSettings settings = null, bool active = false)
+        {
+            if (active)
+            {
+                if (Init(targetLocation))
+                    IFCSettings = settings;
+            }
+        }
+
+        /***************************************************/
+        /**** Private Fields                            ****/
+        /***************************************************/
+
+        private IfcStore m_LoadedModel;
+
+        private Xbim3DModelContext m_3DContext;
+
+
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        private bool Init(string location)
+        {
+            if (string.IsNullOrWhiteSpace(location))
+            {
+                BH.Engine.Reflection.Compute.RecordError("Please specifiy a valid target location.");
+                return false;
+            }
+
+            if (!location.ToLower().EndsWith(".ifc"))
+            {
+                BH.Engine.Reflection.Compute.RecordError("The file needs to be in .ifc format.");
+                return false;
+            }
+
+            try
+            {
+                m_LoadedModel = IfcStore.Open(location);
+            }
+            catch (Exception ex)
+            {
+                BH.Engine.Reflection.Compute.RecordError($"The file failed to load with the following error:\n{ex.Message}");
+                return false;
+            }
+            
+            // By default, the objects are appendend to the file if it exists already.
+            this.m_AdapterSettings.DefaultPushType = oM.Adapter.PushType.CreateOnly;
+
```

Comment:
```suggestion
```

Delete this because you don't support any Push yet.

Reply:
Done